tools/add_triage_signature.py

```diff
@@ -1499,6 +1499,46 @@ def _process_ticket(self, url, issue_key):
             )
 
 
+class FlappingValidations(ErrorSignature):
+    regexps = [
+        re.compile(r"Host .+: validation '.+' is now fixed"),
+        re.compile(r"Host .+: validation '.+' that used to succeed is now failing"),
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. Flapping validations:",
+            label="flapping-validations",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        validation_state_changes = {}
+        cluster = get_metadata_json(url)["cluster"]
+        events_by_host = get_events_by_host(url, cluster["id"])
+        for event in list(itertools.chain(*events_by_host.values())):
+            if any([regexp.match(event["message"]) for regexp in self.regexps]):
+                if event["message"] not in validation_state_changes:
+                    validation_state_changes[event["message"]] = 1
+                else:
+                    validation_state_changes[event["message"]] += 1
```

Comment:
I believe this is equivalent:

```suggestion
        validation_state_changes = defaultdict(int)
        for event in get_events_json(url, get_metadata_json(url)["cluster"]["id"]):
            if any(regexp.match(event["message"]) for regexp in self.regexps):
                validation_state_changes[event["message"]] += 1
```

Reply:
Or even better:

```suggestion
        validation_state_changes = collections.Counter(
            event["message"] for event in
            get_events_json(url, get_metadata_json(url)["cluster"]["id"])
            if any(regexp.match(event["message"]) for regexp in self.regexps)
        )
```

Reply:
Good idea, implemented that